threedi_schema/migrations/versions/0300_upgrade_db_settings.py

```diff
@@ -0,0 +1,296 @@
+    remove_columns_from_table(src_table, [src for src, _ in columns])
+
+
+def move_values_to_table(src_table: str, src_col: str, dst_table: str, dst_col: str):
+    op.execute(f'UPDATE {dst_table} SET {dst_col} = (SELECT {src_col} FROM {src_table} LIMIT 1)')
+    remove_columns_from_table(src_table, [src_col])
+
+
+def remove_columns_from_table(table_name: str, columns: List[str]):
+    # no checks for existence are done, this will fail if any table or column doesn't exist
+    with op.batch_alter_table(table_name) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_settings_id():
+    for settings_col, settings_id, settings_table in GLOBAL_SETTINGS_ID_TO_BOOL:
+        if settings_col != '':
+            # set boolean 'use_*' in model_settings if a relationship exists
+            op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE {settings_id} IS NOT NULL;")
+        # remove all settings rows, exact for the one matching
+        # delete_all_but_matching_id(settings_table, settings_id)
+        op.execute(
+            f"DELETE FROM {settings_table} WHERE NOT EXISTS "
+            f"(SELECT 1 FROM model_settings WHERE model_settings.{settings_id} IS NOT NULL);")
+        # command above doesn't catch id = None, so an extra command is needed for those cases
+        op.execute(
+            f"DELETE FROM {settings_table} WHERE id != "
+            f"(SELECT {settings_id} FROM model_settings);")
+
+    # set use_groundwater_flow
+    sql = """
+        UPDATE model_settings
+        SET use_groundwater_flow = CASE
+            WHEN (SELECT groundwater_hydraulic_conductivity FROM groundwater LIMIT 1) IS NOT NULL OR (SELECT groundwater_hydraulic_conductivity_file FROM groundwater LIMIT 1) IS NOT NULL THEN 1
+            ELSE 0
+        END;
+        """
+    op.execute(sql)
+
+
+def drop_columns(table, columns):
+    with op.batch_alter_table(table) as batch_op:
+        for column in columns:
+            batch_op.drop_column(column)
+
+
+def set_use_from_exists(table, settings_col):
+    op.execute(f"UPDATE model_settings SET {settings_col} = TRUE WHERE EXISTS (SELECT 1 FROM {table});")
+
+
+def delete_all_but_matching_id(table, settings_id):
+    op.execute(f"DELETE FROM {table} WHERE id NOT IN (SELECT {settings_id} FROM model_settings);")
+
+
+def delete_all_but_first_row(table):
+    op.execute(f"DELETE FROM {table} WHERE id NOT IN ("
+               f"SELECT id FROM "
+               f"(SELECT id FROM {table} ORDER BY id LIMIT 1) AS subquery);")
+
+
+def correct_dem_paths():
+    # remove path - this will not work with nested paths!
+    op.execute(f"UPDATE model_settings SET dem_file = SUBSTR(dem_file, INSTR(dem_file, '/') + 1)")
+
+
+def upgrade():
+    conn = op.get_bind()
+    # Only use first row of global settings
+    delete_all_but_first_row("v2_global_settings")
```

Comment:
I double checked, the two places where the "global settings" are used:

grid builder sorts on id and uses first:
https://github.com/nens/threedigrid-builder/blob/0b90c13b1f465d06f96326877e1f7c20cce6c0a1/threedigrid_builder/interface/db.py#L148

simulation template worker only uses one, without explicit sorting. I guess sorting on id is done by default.
https://github.com/nens/threedi-api/blob/44102a7101e107e13a851802aa6570ec6d3d79eb/workers/simulation_templates/simulation_templates/extractor.py#L64

Reply:
Gridbuilder changes should be covered by https://github.com/nens/threedigrid-builder/pull/356 and simulation template worker changes shoud be covered by https://github.com/nens/threedi-api/pull/2190